### PR TITLE
Feat basic routing

### LINKED ISSUE #2: feat-basic-routing

### PR DESCRIPTION
**Pull request by coldpigli**
What's new?

Added Routing
Added Theme Context
Added Sidebar

### CODE REVIEW

File: src/components/SideNav/SideNav.module.css [hunk 1]
```diff
@@ -0,0 +1,61 @@
+.sidenav{
+    padding: var(--space-sm);
+    display: flex;
+    flex-direction: column;
+    align-items: center;
+    justify-content: flex-start;
+    gap: 1.5rem;
+    position: fixed;
+    transition: 200ms ease;
+    /* background-color: var(--background-dark); */
```
Comment (srejitk) [inline src/components/SideNav/SideNav.module.css:10]: Remove this commented code if not needed

Comment (nakulsharma15) [inline src/components/SideNav/SideNav.module.css:10]: Remove the comment if it is no longer required.

**Review by srejitk — APPROVED** [review]
Good starting code. Liked how you implemented your sidenav.
Good luck!

File: src/components/SideNav/SideNav.module.css (hunk 1 above)
Comment (krishanu2911) [inline src/components/SideNav/SideNav.module.css:10]: Remove the comment if not required

**Review by nakulsharma15 — APPROVED** [review]
Overall looks good.
- I liked how nicely you divided things into different components.
- The overall folder structure is quite good and it is easy to locate files in them.
- The naming of the files and functions is also very good and is self-explanatory.

**Review by krishanu2911 — APPROVED** [review]
Well written code 
- Nicely divided things into component
- Liked the theme context

**Pull request merged**